rust/pact_ffi/src/mock_server/handles.rs

```diff
@@ -839,8 +839,12 @@ pub extern fn pactffi_with_query_parameter(
 /// const char* value = "{\"value\":[\"2\"], \"pact:matcher:type\":\"regex\", \"regex\":\"\\\\d+\"}";
 /// pactffi_with_query_parameter_v2(handle, "id", 0, value);
 /// ```
+///
+/// For a query parameter with no value, the value parameter can be set to a NULL pointer.
+///
 /// # Safety
-/// The name and value parameters must be valid pointers to NULL terminated strings.
+/// The name parameter must be a valid pointer to a NULL terminated string. If the value
+/// parameter is not NULL, it must point to a valid NULL terminated string.
```

Comment:
I think it would be good to make the documentation more explicit, with perhaps some example usage to distinguish an "empty value" from a "null".

E.g.:

---

For query parameters with no value, two distinct formats are provided:

1. Parameters with blank values, as specified by `?foo=&bar=`, require an empty string:

   ```c
   pactffi_with_query_parameter_v2(handle, "foo", 0, "");
   pactffi_with_query_parameter_v2(handle, "bar", 0, "");
   ```

2. Parameters with no associated valued, as specified by `?foo&bar`, require a `NULL` pointer:

   ```c
   pactffi_with_query_parameter_v2(handle, "foo", 0, NULL);
   pactffi_with_query_parameter_v2(handle, "bar", 0, NULL);
   ```

Note that the two formats are distinct; however, support for either variant can be provided through the integration JSON:

```c
const char* value = "...";
pactffi_with_query_parameter_v2(handle, "foo", 0, value);
```

Reply:
I will copy your example docs :-D

Reply:
Nice, the only one to fill out would be the last one (as I didn't bother filling out the Integration JSON text) 👍

Reply:
Integration JSON won't support it, it will always resolve to the empty string.